flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/adapter/DefaultResultPartitionTest.java

```diff
@@ -0,0 +1,183 @@
+		executionGraph = createSimpleTestGraph(
+			new JobID(),
+			taskManagerGateway,
+			triggeredRestartStrategy,
+			jobVertices);
+		adapter = new ExecutionGraphToSchedulingTopologyAdapter(executionGraph);
+
+		intermediateResultPartitions = new ArrayList<>();
+		schedulingResultPartitions = new ArrayList<>();
+
+		for (ExecutionVertex vertex : executionGraph.getAllExecutionVertices()) {
+			for (Map.Entry<IntermediateResultPartitionID, IntermediateResultPartition> entry
+				: vertex.getProducedPartitions().entrySet()) {
+				intermediateResultPartitions.add(entry.getValue());
+				schedulingResultPartitions.add(adapter.getResultPartition(entry.getKey())
+					.orElseThrow(() -> new IllegalArgumentException("can not find partition" + entry.getKey())));
+			}
+		}
+		assertEquals(parallelism, intermediateResultPartitions.size());
+	}
+
+	@Test
+	public void testBasicInfo() {
+		for (int idx = 0; idx < intermediateResultPartitions.size(); idx++) {
+			final IntermediateResultPartition partition = intermediateResultPartitions.get(idx);
+			final SchedulingResultPartition schedulingResultPartition = schedulingResultPartitions.get(idx);
+			assertEquals(partition.getPartitionId(), schedulingResultPartition.getId());
+			assertEquals(partition.getIntermediateResult().getId(), schedulingResultPartition.getResultId());
+			assertEquals(partition.getResultType(), schedulingResultPartition.getPartitionType());
+		}
+	}
+
+	@Test
+	public void testGetConsumers() {
+		for (int idx = 0; idx < intermediateResultPartitions.size(); idx++) {
+			Collection<ExecutionVertexID> schedulingConsumers = schedulingResultPartitions.get(idx).getConsumers()
+				.stream().map(SchedulingExecutionVertex::getId).collect(Collectors.toList());
+
+			Set<ExecutionVertexID> executionConsumers = new HashSet<>();
+			for (List<ExecutionEdge> list :intermediateResultPartitions.get(idx).getConsumers()) {
+				for (ExecutionEdge edge : list) {
+					final ExecutionVertex vertex = edge.getTarget();
+					executionConsumers.add(new ExecutionVertexID(vertex.getJobvertexId(), vertex.getParallelSubtaskIndex()));
+				}
+			}
+			assertThat(schedulingConsumers, containsInAnyOrder(executionConsumers.toArray()));
+		}
+	}
+
+	@Test
+	public void testGetProducer() {
+		for (int idx = 0; idx < intermediateResultPartitions.size(); idx++) {
+			final ExecutionVertex vertex = intermediateResultPartitions.get(idx).getProducer();
+			assertEquals(schedulingResultPartitions.get(idx).getProducer().getId(),
+				new ExecutionVertexID(vertex.getJobvertexId(), vertex.getParallelSubtaskIndex()));
+		}
+	}
+
+	@Test
+	public void testGetPartitionState() {
+		List<SchedulingExecutionVertex> schedulingExecutionVertices = new ArrayList<>();
+		executionGraph.getAllExecutionVertices().forEach(
+			vertex -> schedulingExecutionVertices.add(new DefaultExecutionVertex(vertex)));
+
+		final ExecutionState[] states = ExecutionState.values();
+		Random random = new Random();
```

Comment:
same as in the other test, the random element is not necessary imo

Reply:
OK